### PR TITLE
Add search functionallity with Fuse.js

### PR DESCRIPTION
**Pull request by dantehemerson**
This solves #178
Preview: https://howoldisit-preview.netlify.com
### Features:
* Improve search functionality with Fuse.js.
* Now the search is more fluid when the letters are typed quickly.
* Also separate the application into more components such as: Header, List (list of technologies), Footer and more.

### CODE REVIEW

**Comment by dantehemerson** [comment]
WHEN YOU LOOK FOR ANY TECHNOLOGY APPEARS DOES NOT APPEAR FIRST?

An example is when we look for `redux`, it appears third in the results when it is ordered from **A-Z**.

**Fusejs** has a very intelligent way of searching (not just comparing the first letters of the word).
What is happening now is that the results are modified by including filters.

So when we look for `redux` it does not appear first since we are sorting by name or by age (which is bad for me).
My suggestion is to remove the filters when the search is done, and only show it when it is not looking for.

Do you think I should remove the filters or keep them that way?

**Comment by Esteban-Rocha** [comment]
Hey Thank you for your work on this! :clap: I got however some feedback and doubts so maybe you could help me sort this out.

Testing your [preview link](howoldisit-preview.netlify.com) I got this behavior:

image

Maybe I'm missing something? but shouldn't this act like it's currently working by hiding all other results? ->

image

Besides of that I could not reproduce #178, as the app is now it works quite fast for me, so maybe we should get more feedback from other users to validate this?

**Comment by dantehemerson** [comment]
I'm sorry for the delay.

Just that I was referring above, if you notice there is sorting in alphabetical order (A-Z), and the results are sorted that way. That is the error.
If we do not order the results that Fusejs gives us, then `bash` would come out as the first result.
There are many results because Fusejs looks for similar words (I think I should have configured the sensitivity)

For example if we had an typographic error:
`jvscript`
the result would be
```
Javascript
Typescript
Nativescript
```
in that order.

Basically those are the benefits of using Fusejs.

**Comment by dantehemerson** [comment]
Regarding the speed of the search input, personally for me it works slow. I have an core i3 computer with 4gb ram. Probably just my problem.

@jsrn @rithvikvibhu You have the error of the issue #178? I would like to resolve this doubt.
If not, we could close this PR because it was based on the problem of input #178

Now if you want to implement Fusejs.  I would like to help.

**Comment by rithvikvibhu** [comment]
I think fusejs is slightly faster, but where it really shines is with fuzzy search (even searching with spelling mistakes and such).
Although, like @dantehemerson said, reducing the sensitivity would be better.
